tools/lib/bootpool.js

```diff
@@ -0,0 +1,299 @@
+                        mountpoint));
+                    return;
+            }
+
+            status.steps.marker_file = marker;
+            callback(null, status);
+        });
+    });
+}
+
+/*
+ * Mount bootfs onto the default mountpoint.
+ */
+function
+ensure_bootfs_mounted(poolname, callback)
+{
+    var args = [];
+    var mountpoint;
+
+    mod_assert.func(callback, 'callback');
+    mod_assert.string(poolname, 'poolname');
+
+    dprintf('ensuring bootfs is mounted (no altmountopts for bootfs)...\n');
+
+    args.push('list');
+    args.push('-Ho');
+    args.push('bootfs');
+    args.push(poolname);
+    mod_child.execFile('zpool', args, function (err, stdout, stderr) {
+        if (err) {
+            callback(new VError(err, 'Cannot find bootfs for %s', poolname),
+                     '');
+            return;
+        }
+
+        lib_usbkey.get_mountpoints(function (err, mtpts) {
+            if (err) {
+                callback(new VError(err, 'could not read mount configuration'));
+                return;
+            }
+
+            /* Only use default mountpoints. */
+            mountpoint = mtpts[0];
+
+            /* Make sure the mountpoint is there. */
+            lib_usbkey.ensure_mountpoint_exists(mountpoint, function (err) {
+                if (err) {
+                    callback(err, '');
+                    return;
+                }
+                var bootfs = '/' + stdout.trim();
+                /* Check for /pool/boot standard? */
+                if (bootfs !== '/' + poolname + '/boot') {
+                    dprintf('HEADS UP: bootfs %s is not SmartOS-standard.\n');
+                }
+
+                /*
+                 * See if anything is mounted in mountpoint already, and
+                 * umount it.
+                 */
+                dprintf('fetching pool mount ensure information for "%s"\n',
+                        mountpoint);
+
+                function do_lofs_mount(bootfs, mountpoint, callback) {
+                    /* Okay, let's lofs-mount bootfs on to mountpoint. */
+                    var lofsargs = [];
+
+                    dprintf('lofs-mounted bootpool path "%s" on "%s".\n',
+                        bootfs, mountpoint);
+
+                    lofsargs.push('-F');
+                    lofsargs.push('lofs');
+                    lofsargs.push(bootfs);
+                    lofsargs.push(mountpoint);
+                    mod_child.execFile('mount', lofsargs, function (err,
+                        stdout, stderr) {
+                        if (err) {
+                            callback(new VError(err, 'Cannot lofs mount %s',
+                                bootfs));
+                        } else {
+                            callback(null, mountpoint);
+                        }
+                        return;
+                    });
+                }
+
+                lib_usbkey.get_mount_info(mountpoint, function (err, mi) {
```

Comment:
Give anon func a name.

Reply:
Someone else asked me about that earlier, and I went by what the old code did.  Is naming callbacks a stylistic convention we need to catch up with?  If so, we should file a bug to make it ALL good this way.